File: back-end/Fretefy.Test.Domain/Dto/Cidade/CidadeResponseDto.cs
using Fretefy.Test.Domain.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace Fretefy.Test.Domain.Dto
{
    public class CidadeResponseDto
    {
        public Guid Id { get; set; }
        public string Nome { get; set; }
        public string UF { get; set; }
        public List<RegiaoCidade> regiaoCidades { get; set; }
    }
}

File: front-end/src/app/models/dto/cidade/cidade/cidade-response-dto.ts
import { RegiaoCidade } from '../../../regiao-cidade';

export class CidadeResponseDto {
  id: string;
  nome: string;
  uf: string;
  regiaoCidades: RegiaoCidade[];

  constructor(id: string, nome: string, uf: string, regiaoCidades: RegiaoCidade[]) {
    this.id = id;
    this.nome = nome;
    this.uf = uf;
    this.regiaoCidades = regiaoCidades;
  }
}